### PR TITLE
Add or remove from lists in AccountActivity

### LINKED ISSUE #2815: Request - Add to List from Profile
It would be incredibly helpful if a profile could be added to a List directly from the profile's page (via the ... menu would make sense), as is possible using the Mastodon advanced web interface.

Screenshot_20221112_092922

* * * *
- Tusky Version: 19.0
- Android Version: 12
- Android Device: Samsung Galaxy S20+

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

### PR DESCRIPTION
**Pull request by kyori19**
Implemented "Add or remove from list" feature in account column of Mastodon frontend.
Though Mastodon doesn't show this menu for self account, I configured to show since Mastodon supports adding self into lists.

* Fix #2815 
* Fix accelforce/Yuito#20

# Questions

* I don't know why but text in the Snackbar is really hard to see. Please help me if you know how to fix this.
* Currently spawned Snackbar disappears automatically after `LENGTH_LONG`. Should this be `LENGTH_INDEFINITE`?

# Screenshots

image
image
image
image

### CODE REVIEW

File: app/src/main/res/layout/item_add_or_remove_from_list.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        android:id="@+id/listNameView"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_weight="1"
+        android:drawablePadding="8dp"
+        android:ellipsize="end"
+        android:gravity="center_vertical"
+        android:maxLines="1"
+        android:textSize="?attr/status_text_medium"
+        app:drawableTint="?android:attr/textColorSecondary"
```
Comment (connyduck) [inline app/src/main/res/layout/item_add_or_remove_from_list.xml:22]: should be same color as text (probably best to set textColor to the same value)

**Review by connyduck — COMMENTED** [review]
> I don't know why but text in the Snackbar is really hard to see. Please help me if you know how to fix this.

It is because you use `TuskyDialogFragmentStyle` which overrides`android:backgroundTint`. I don't know why there was a need to override the backgroundTint, it looks like this line can be removed and then the snackbar looks correct.

> Currently spawned Snackbar disappears automatically after LENGTH_LONG. Should this be LENGTH_INDEFINITE?

I think LENGTH_LONG is fine.

**Comment by connyduck** [comment]
A "done" button to easily dismiss the dialog fragment would be cool

**Comment by kyori19** [comment]
✔️ Fixed

**Review by connyduck — APPROVED** [review]
Will merge after Tusky 20 release

**Comment by thrillfall** [comment]
@kyori19 thanks for contributing this. That was one big thing I was missing in this app

**Comment by fadookie** [comment]
Thank you! I was starting to work on this feature myself because I wanted it so badly, happy to find out it's already done!

I built a test build for myself off of this branch and will try to start using this feature, if I run into any issues I'll let you know.

**Pull request merged**